beetsplug/lyrics.py

```diff
@@ -144,29 +145,37 @@ def search_pairs(item):
     The method also tries to split multiple titles separated with `/`.
     """
 
+    def strip_part(string, pattern):
+        """Return first matching group if string matches pattern, the full
+           string otherwise."""
+        match = re.search(pattern, string, re.IGNORECASE)
+        if match:
+            return match.group(1)
+        return string
+
     title, artist = item.title, item.artist
     titles = [title]
     artists = [artist]
 
     # Remove any featuring artists from the artists name
-    pattern = r"(.*?) {0}".format(plugins.feat_tokens())
-    match = re.search(pattern, artist, re.IGNORECASE)
-    if match:
-        artists.append(match.group(1))
+    artists.append(
+        strip_part(artist, r"(.*?) {0}".format(plugins.feat_tokens())))
 
     # Remove a parenthesized suffix from a title string. Common
     # examples include (live), (remix), and (acoustic).
-    pattern = r"(.+?)\s+[(].*[)]$"
-    match = re.search(pattern, title, re.IGNORECASE)
-    if match:
-        titles.append(match.group(1))
+    titles.append(strip_part(title, r"(.+?)\s+[(].*[)]$"))
 
     # Remove any featuring artists from the title
     pattern = r"(.*?) {0}".format(plugins.feat_tokens(for_artist=False))
-    for title in titles[:]:
-        match = re.search(pattern, title, re.IGNORECASE)
-        if match:
-            titles.append(match.group(1))
+    for title in list(titles):
+        titles.append(strip_part(title, pattern))
+
+    # Remove part of the title string after colon ':' for songs with subtitles
+    titles.append(strip_part(title, r"(.+?)\s*:.*"))
+
+    # Yield artist and title obtained from item first
+    titles = list(OrderedDict.fromkeys([title] + titles))
+    artists = list(OrderedDict.fromkeys([artist] + artists))
```

Comment:
I moved the burden to update the alternatives lists to the function that performs the `re.search` and think it works better that way : it avoids the *return None+filter* processing and makes caller code simpler.